include/bencode.hpp

```diff
@@ -258,16 +258,16 @@ namespace bencode {
     template<typename Integer>
     inline void check_overflow(Integer value, Integer digit) {
       using limits = std::numeric_limits<Integer>;
-      if((value > limits::max() / 10) ||
-         (value == limits::max() / 10 && digit > limits::max() % 10))
+      if((value > (limits::max)() / 10) ||
+         (value == (limits::max)() / 10 && digit > (limits::max)() % 10))
```

Comment:
This is just to work around how Windows headers define `max` and `min` as preprocessor macros, right? I generally prefer to recommend using `#define NOMINMAX` instead. Jumping through these hoops to work around a mistake Microsoft made decades ago is pretty annoying, although it's not too big a deal in this particular case.

I'll have to think about this for a bit. While I don't think I'd accept something like this in a more-complicated project, this is a fairly simple change and doesn't impact readability too badly. I'll try to get back with a decision sometime next week.

Reply:
Hello, I have a test on Visual Studio 2019. If you'd like a conditional directive, it will looks like this:

`
#if defined(_MSC_VER)
#undef min
#undef max
#endif
`